service/src/main/java/bio/terra/externalcreds/dataAccess/SshKeyPairDAO.java

```diff
@@ -91,6 +93,25 @@ public SshKeyPairInternal upsertSshKeyPair(SshKeyPairInternal sshKeyPairInternal
         Objects.requireNonNull(generatedKeyHolder.getKey()).intValue());
   }
 
+  /** Gets expired or un-encrypted ssh key pair. */
+  public List<SshKeyPairInternal> getExpiredOrUnEncryptedSshKeyPair() {
+    var kmsConfig = externalCredsConfig.getKmsConfiguration();
+    if (kmsConfig == null) {
+      return Collections.emptyList();
+    }
+    var namedParameters =
+        new MapSqlParameterSource(
+            "refreshCutoffTimestamp",
+            Timestamp.from(Instant.now().minus(kmsConfig.getSshKeyPairRefreshDuration())));
```

Comment:
I would prefer this value to be passed in as a parameter. It is business logic to determine the cutoff. It also makes this function easier to test.